src/key.rs

```diff
@@ -20,6 +20,8 @@
 #[cfg(any(test, feature = "rustc-serialize"))] use serialize::{Decoder, Decodable, Encoder, Encodable};
 #[cfg(any(test, feature = "serde"))] use serde::{Serialize, Deserialize, Serializer, Deserializer};
 
+use std::mem;
```

Comment:
This would not be needed if you'd write `let mut ret = ffi::PublicKey::new();` below.

Reply:
Matt points out that `PublicKey::new()` creates a public key which will trigger an assertion failure in a lot of secp functions if it's actually used. So that function is itself unsafe and shouldn't have been exposed.

Going forward I either need to deprecate the function or audit everything else to check for the blank pubkey before calling into libsecp. Not sure what the most appropriate option is, but in the meantime I'm just going to avoid it here.